velox/functions/sparksql/aggregates/BloomFilterAggAggregate.cpp

```diff
@@ -144,17 +136,18 @@ class BloomFilterAggAggregate : public exec::Aggregate {
     accumulator->init(capacity_);
     if (decodedRaw_.isConstantMapping()) {
       // All values are same, just do for the first.
-      checkBloomFilterNotNull(decodedRaw_, 0);
-      accumulator->insert(decodedRaw_.valueAt<int64_t>(0));
-      return;
-    }
-    auto mayHaveNulls = decodedRaw_.mayHaveNulls();
-    rows.applyToSelected([&](vector_size_t row) {
-      if (mayHaveNulls) {
-        checkBloomFilterNotNull(decodedRaw_, row);
+      if (!decodedRaw_.isNullAt(0)) {
+        accumulator->insert(decodedRaw_.valueAt<int64_t>(0));
       }
-      accumulator->insert(decodedRaw_.valueAt<int64_t>(row));
-    });
+    } else {
+      auto mayHaveNulls = decodedRaw_.mayHaveNulls();
```

Comment:
Seems we don't need to check `mayHaveNulls` for each input. Could be separated as below.
```
  const auto mayHaveNulls = decodedRaw_.mayHaveNulls();
  if UNLIKELY (mayHaveNulls) {
    rows.applyToSelected([&](vector_size_t row) {
      if UNLIKELY (decodedRaw_.isNullAt(row)) {
        return;
      }
      accumulator->insert(decodedRaw_.valueAt<int64_t>(row));
    }
    return;
  }
  rows.applyToSelected([&](vector_size_t row) {
    accumulator->insert(decodedRaw_.valueAt<int64_t>(row));
  }
```